### PR TITLE
(refactor) Make `Runtime` class synchronous

### PR DESCRIPTION
**Pull request by xingyaoww**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Previously we make `Runtime` class async, only because we want to use `aiohttp.Session`, which can be easily replaced by `request.Session` 😅. We didn't fully utilize features of async, but instead added a lot of burden to ourselves -- the async runtime breaks the current multi-processing evaluation and make it harder for people interested in using our Runtime: they would have to make their code async before they are able to import openhands.runtime and start using it.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

* Replace `aiohttp.Session` with `request.Session`
* Support using Runtime as a context manager
* Fix all runtime occurrences in testcases
* Fix all runtime occurrences in evaluation
* Fix running evaluation in parallel
* Support running evaluation in parallel using the Runtime API

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by tobitege** [comment]
This might block going forward to get async communication between UI and backend?
Like simultaneous terminal updates while waiting on an LLM reply etc.

**Comment by xingyaoww** [comment]
@tobitege I thought about it too - I think we can achieve the same thing with `StreamableObservation` similar to HTTP streaming, which doesn't mandate async if I understand it correctly?

I imagine we can do something like this: https://cookbook.openai.com/examples/how_to_stream_completions#2-how-to-stream-a-chat-completion

**Review by neubig — APPROVED** [review]
Looks fine to me! There were lots of changes but they were all pretty straightforward

File: openhands/runtime/client/client.py [hunk 1]
```diff
@@ -511,7 +511,9 @@ async def execute_action(action_request: ActionRequest):
 
     @app.post('/upload_file')
     async def upload_file(
-        file: UploadFile, destination: str = '/', recursive: bool = False
+        file: UploadFile,
```
[the remark this answers is not in the document]
Comment (xingyaoww) [reply openhands/runtime/client/client.py:514]: This was my attempt to debug some weird file upload bug (but it turns out this was not the source of bug)-- reverted!

**Pull request merged**